exercises/react/src/lib.rs

```diff
@@ -0,0 +1,91 @@
+#[allow(unused_variables)]
+
+// Because these are passed without & to some functions,
+// it will probably be necessary for these two types to be Copy.
+pub type CellID = ();
+pub type CallbackID = ();
+
+pub struct Reactor<T> {
+    // Just so that the compiler doesn't complain about an unused type parameter.
+    // You probably want to delete this field.
+    dummy: T,
+}
+
+// You are guaranteed that Reactor will only be tested against types that are Copy + PartialEq.
```

Comment:
> If T is not `Copy` and `create_input` still passes a `T`:
> 
> * I think the cell can take ownership for `value`, and a reference for `last_value`.

Illegal.

```
src/lib.rs:23:26: 23:33 error: `initial` does not live long enough
src/lib.rs:23             last_value: &initial,
                                       ^~~~~~~
src/lib.rs:21:60: 30:6 note: reference must be valid for the lifetime 'a as defined on the block at 21:59...
src/lib.rs:21     fn new(initial: T, cell_type: CellType<'a, T>) -> Self {
                                                                         ^
src/lib.rs:21:60: 30:6 note: ...but borrowed value is only valid for the scope of function body at 21:59
src/lib.rs:21     fn new(initial: T, cell_type: CellType<'a, T>) -> Self {
```

Of course. Since `initial` is the parameter to that function. Doh.